medusa/restore_node.py

```diff
@@ -225,10 +228,19 @@ def table_is_allowed_to_restore(keyspace, table, fqtns_to_restore):
     return True
 
 
-def clean_path(p):
-    if p.exists():
-        logging.debug('Cleaning ({})'.format(p))
-        subprocess.check_output(['sudo', '-u', p.owner(), 'rm', '-rf', str(p)])
+def clean_path(p, keep_folder=False):
+    path = str(p)
+    if p.exists() and os.path.isdir(path) and len(os.listdir(path)):
+        logging.debug('Cleaning ({})'.format(path))
+        if keep_folder:
+            logging.debug('Removing files - keep folder {}'.format(path))
+            for f in os.listdir(path):
+                file_path = os.path.join(path, f)
+                logging.debug('Removing file {}'.format(file_path))
+                os.remove(file_path)
```

Comment:
Why not use the `rm` command with subprocess here as well? Or the other way around so we're consistent in how we access the FS. Either all through Python modules (more portable code) or with shell commands (easier to deal with file ownership?).

Reply:
Makes sense, I thought about that, and dropped the ball. 100% agree on making this consistent, and we use bash commands everywhere else 👍.

Reply:
Here is the diff, I'll make sure this works.

```
-                os.remove(file_path)
+                subprocess.check_output(['sudo', '-u', p.owner(), 'rm', '-rf', file_path])
```